ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -85,6 +90,23 @@ public UIManagerModule(
     super(reactContext);
     mEventDispatcher = new EventDispatcher(reactContext);
     DisplayMetrics displayMetrics = reactContext.getResources().getDisplayMetrics();
+
+    // Get the real display metrics if we are using API level 17 or higher.
+    // The real metrics include system decor elements (e.g. soft menu bar).
+    //
+    // See: http://developer.android.com/reference/android/view/Display.html#getRealMetrics(android.util.DisplayMetrics)
+    if (Build.VERSION.SDK_INT >= 17) {
+      WindowManager wm = (WindowManager)reactContext.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
+
+      try {
+        Display display = wm.getDefaultDisplay();
+        Display.class.getMethod("getRealMetrics", DisplayMetrics.class).invoke(display, displayMetrics);
+      } catch (InvocationTargetException e) {
+      } catch (IllegalAccessException e) {
+      } catch (NoSuchMethodException e) {
```

Comment:
OK, I can add a log. There isn't much you can do with reflection exceptions. 

Should be able to return wrapped invocation exception. 

Is there a best practice for rethrow in RN?